jpa/org.eclipse.persistence.jpa/src/test/java/org/eclipse/persistence/jpa/PersistenceProviderTest.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+package org.eclipse.persistence.jpa;
+
+import java.util.Map;
+
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.PersistenceConfiguration;
+import jakarta.persistence.spi.PersistenceUnitInfo;
+import jakarta.persistence.spi.ProviderUtil;
+import org.eclipse.persistence.config.PersistenceUnitProperties;
+import org.eclipse.persistence.internal.jpa.EntityManagerFactoryProvider;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class PersistenceProviderTest {
+
+    @Test
+    void testCheckForProviderPropertyWithoutProperty() {
+        Map<String, Object> properties = Map.of();
+        boolean result = PersistenceProvider.checkForProviderProperty(properties);
```

Comment:
What about tests based on `Persistence.createEntityManagerFactory(` to avoid `public` scope for `checkForProviderProperty` and others. Something like `org.eclipse.persistence.testing.tests.jpa.advanced.EntityManagerJUnitTestSuite#testProviderPropertySetting`.

Reply:
`checkForProviderProperty` WAS public before my changes. There is no relation between this test and method visibility.